examples/distributed_autofaiss_n_indices.py

```diff
@@ -0,0 +1,56 @@
+"""
+An example of running autofaiss by pyspark to produce N indices.
+
+You need to install pyspark before using the following example.
+"""
+from typing import Dict
+
+import faiss
+import numpy as np
+
+from autofaiss import build_index
+
+# You'd better create a spark session before calling build_index,
+# otherwise, a spark session would be created by autofaiss with the least configuration.
+
+index_path2_metric_infos: Dict[str, Dict] = build_index(
+    embeddings="hdfs://root/path/to/your/embeddings/folder",
+    distributed="pyspark",
+    file_format="parquet",
+    temporary_indices_folder="hdfs://root/tmp/distributed_autofaiss_indices",
+    current_memory_available="10G",
+    max_index_memory_usage="100G",
+    nb_indices_to_keep=10,
+)
+
+index_paths = sorted(index_path2_metric_infos.keys())
+
+###########################################
+# Use case 1: merging 10 indices into one #
+###########################################
+merged = faiss.read_index(index_paths[0])
+for rest_index_file in index_paths[1:]:
+    index = faiss.read_index(rest_index_file)
+    faiss.merge_into(merged, index, shift_ids=True)
+with open("merged-knn.index", "wb") as f:
+    faiss.write_index(merged, faiss.PyCallbackIOWriter(f.write))
+
+########################################
+# Use case 2: searching from N indices #
+########################################
+K, DIM, all_distances, all_ids, NB_QUERIES = 5, 512, [], [], 2
+queries = faiss.rand((NB_QUERIES, DIM))
+for rest_index_file in index_paths:
+    index = faiss.read_index(rest_index_file)
+    distances, ids = index.search(queries, k=K)
+    all_distances.append(distances)
+    all_ids.append(ids)
```

Comment:
this is incorrect because the ids are not consecutive over all indices

Reply:
Ah yes, good catch !